### PR TITLE
fix win compiler error for mkl sparse matrix

### PR DESCRIPTION
**Pull request by triplekings**
## Description ##
fix compiler error for windows cmake.
https://github.com/apache/incubator-mxnet/issues/14730

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by pengzhao-intel** [comment]
Currently, the MKL implementation and the original one is mixed which makes the thing complicated and hard to get rid of one concise :(

I suggest writing all MKL related code into a separated file and we can import it in `dot-imp.h` only under the right situation.

**Comment by triplekings** [comment]
> Thanks for your PR. In the title you say CMake, but you are changing the Makefile. Could you clarify?
> 
> Are you compiling in windows with MKL?
Thanks for your reminder. Actually, I just fix it on win platform no matter CMAKE or Makefile. The issue was raised on WIN cmake, so it works

File: Makefile [hunk 1]
```diff
@@ -412,8 +412,8 @@ ifeq ($(USE_DIST_KVSTORE), 1)
 endif
 
 #sparse-matrix
+SPARSE_MATRIX_DIR =  $(ROOTDIR)/3rdparty/sparse-matrix
 ifeq ($(USE_BLAS), mkl)
-	SPARSE_MATRIX_DIR =  $(ROOTDIR)/3rdparty/sparse-matrix
```
Comment (larroy) [inline Makefile]: I thought we only need sparse-matrix when on MKL, is that not correct?

Comment (triplekings) [reply Makefile]: yes, you are right.
I just move it back under MKL macro.

**Comment by yinghu5** [comment]
@triplekings.  I wonder if the fix trigger some other windows issues.   I tested the fix about incubator-mxnet\src\operator\tensor\./dot-inl.h (dot.cc)  under windows MSVC 2015.   it raises some new build errors. like

              ] (compiling source file C:\mxnet_4\incubator-mxnet\src\operator\tensor\dot.cc)
    13>C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xutility(2364): error C2938: '_Iter_cat_t<int>' : Failed to special
       ize alias template (compiling source file C:\mxnet_4\incubator-mxnet\src\operator\tensor\dot.cc) [C:\mxnet_4\incubator-mxnet\build
       \mxnet.vcxproj]
    13>C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xutility(2364): error C2062: type 'unknown-type' unexpected (compil
       ing source file C:\mxnet_4\incubator-mxnet\src\operator\tensor\dot.cc) [C:\mxnet_4\incubator-mxnet\build\mxnet.vcxproj]
 
Reproduce steps: 
C:\mxnet_4\incubator-mxnet\build>cmake -G "Visual Studio 14 Win64" .. -DUSE_CUDA=0 -DUSE_CUDNN=0 -DUSE_NVRTC=0 -DUSE_OPENCV=1 -DUSE_OPENMP=1 -DUSE_PROFILER=1 -DUSE_BLAS=open -DUSE_LAPACK=1 -DUSE_DIST_KVSTORE=0 -DCUDA_ARCH_NAME=All -DCMAKE_BUILD_TYPE=Release

msbuild mxnet.sln /p:Configuration=Release;Platform=x64 /maxcpucount

Thanks

**Comment by roywei** [comment]
@mxnet-label-bot add[MKL, Sparse]

**Comment by pengzhao-intel** [comment]
We will revert the mkl sparse PR by #14806, so closing this one.

**Pull request closed without merging**